src/services/navigationBar.ts

```diff
@@ -691,4 +703,16 @@ namespace ts.NavigationBar {
                 return false;
         }
     }
+
+    function cleanText(text: string): string {
+        // Truncate to maximum amount of characters as we don't want to do a big replace operation.
+        text = text.length > maxLength ? text.substring(0, maxLength) + "..." : text;
+
+        // Replaces ECMAScript line terminators and removes the trailing `\` from each line:
+        // \n - Line Feed
+        // \r - Carriage Return
+        // \u2028 - Line separator
+        // \u2029 - Paragraph separator
+        return text.replace(/\\?(\r?\n|\u2028|\u2029)/g, "");

```

Comment:
This is wrong regexp. This will change text `"a1\\r\nb"` to `a1`

Reply:
I'm not sure what scenario you're looking at. I executed the following commands on Chrome's console and this is the result:
```js
> 'a1\\r\nb'.replace(/\\?(\r?\n|\u2028|\u2029)/g, "");
"a1\rb"

> 'a1\
\r\nb'.replace(/\\?(\r?\n|\u2028|\u2029)/g, "");
"a1b"

> 'a1\
b'.replace(/\\?(\r?\n|\u2028|\u2029)/g, "");
"a1b"
```

Is there any scenario I'm missing?

Reply:
eg a1\rb is incorrrect because it would be displayed as a1... so replacing \\ before r is incorrect

Reply:
I see, the `\r` by itself doesn't get replaced. It's a weird scenario to consider but I guess it can happen. I just fixed the regex. With it's new change the scenario looks like this: 
```js
> `a1\rb`.replace(/\\?(\r?\n|\r|\u2028|\u2029)/g, "")
"a1b"
```